apps/confidential/confidential.go

```diff
@@ -286,6 +286,9 @@ type Options struct {
 	AzureRegion string
 
 	capabilities []string
+
+	// Default is Instance discovery is enabled, unless disabled by setting DisableInstanceDiscovery = true
+	DisableInstanceDiscovery bool
```

Comment:
Same thought here: applications have no reason to set this field, so we have no reason to export it
```suggestion
	disableInstanceDiscovery bool
```